plugin/withKlaviyoIos.ts

```diff
@@ -142,6 +144,136 @@ const withRemoteNotificationsPermissions: ConfigPlugin<KlaviyoPluginIosProps> =
   });
 };
 
+/**
+ * Logs geofencing configuration status.
+ * Location permission strings are handled by expo-location, not this plugin.
+ */
+const withGeofencingConfig: ConfigPlugin<KlaviyoPluginIosProps> = (
+  config,
+  props
+) => {
+  KlaviyoLog.log('Checking geofencing configuration...');
+  KlaviyoLog.log('Geofencing props: ' + JSON.stringify(props.geofencing));
+  
+  if (!props.geofencing?.enabled) {
+    KlaviyoLog.log('Geofencing is not enabled, skipping geofencing configuration...');
+  } else {
+    KlaviyoLog.log('Geofencing is enabled. Location permissions should be configured via expo-location.');
+  }
+
+  return config;
+};
+
+/**
+ * Injects KlaviyoLocation dependency into the podspec and adds location background mode if geofencing is enabled.
+ */
+const withGeofencingPodspec: ConfigPlugin<KlaviyoPluginIosProps> = (config, props) => {
+  const geofencingEnabled = props.geofencing?.enabled ?? false;
+  
+  // Handle podspec and Swift file modifications
+  config = withDangerousMod(config, [
+    'ios',
+    async config => {
+      const pluginRoot = getPluginRoot();
+      const podspecPath = path.join(pluginRoot, 'ios', 'ExpoKlaviyo.podspec');
+      const swiftPath = path.join(pluginRoot, 'ios', 'ExpoKlaviyo', 'KlaviyoAppDelegate.swift');
+      
+      if (geofencingEnabled) {
+        // Inject KlaviyoLocation dependency into podspec
+        try {
+          let podspecContent = await FileManager.readFile(podspecPath);
+          podspecContent = podspecContent.replace(
+            "# KLAVIYO_LOCATION_DEPENDENCY",
+            "s.dependency 'KlaviyoLocation' # KLAVIYO_LOCATION_DEPENDENCY"
+          );
+          await FileManager.writeFile(podspecPath, podspecContent);
+          KlaviyoLog.log('Injected KlaviyoLocation dependency into podspec');
+        } catch (err) {
+          KlaviyoLog.log('Could not configure podspec for geofencing: ' + err);
+        }
+        
+        // Inject import and registerGeofencing into KlaviyoAppDelegate.swift
+        try {
+          let swiftContent = await FileManager.readFile(swiftPath);
+          swiftContent = swiftContent.replace(
+            "// KLAVIYO_GEOFENCING_IMPORT",
+            "import KlaviyoLocation // KLAVIYO_GEOFENCING_IMPORT"
```

Comment:
nit 
```suggestion
            "import KlaviyoLocation"
```

mostly worried that if something gets regenerated and we add the same hook that this gets double imported